### PR TITLE
Create a Zephyr Test Cycle for all Test Executions

### LINKED ISSUE #32: Create a Zephyr Test Cycle for all Test Executions
We want all test executions in the pipeline to be contained in Zephyr Test Cycles (when the Zephyr capability is activated).

The test cycle shall be:
- named by the environment we are operating in: `D`/`Q`/`P`
- the version should be set to the value in `MROPipelineUtil.getBuildParams().version`
- the build number value in the cycle should be filled with the pipeline run number

Note:
- a test cycle with the same version may already exist (as a pipeline may run multiple times for each version)

### PR DESCRIPTION
**Pull request by metmajer**
Closes #32

### CODE REVIEW

**Comment by metmajer** [comment]
@hrcornejo I have just tested this internally. Here's what I have found: the version shows UNDEFINED and the build number is not yet set. Let's walk through this together in the next days.

File: src/org/ods/usecase/JiraUseCaseZephyrSupport.groovy [hunk 1]
```diff
@@ -4,24 +4,41 @@ import com.cloudbees.groovy.cps.NonCPS
...
         if (!this.zephyr) return
 
+        def testCycleId = "-1"
+        if(!jiraTestIssues?.isEmpty()) {
+            def projectId = jiraTestIssues.first()?.projectId ?: ""
+            def versionId = this.getVersionId(projectId)
+            def cycleName = this.util.getBuildParams()?.targetEnvironmentToken
+            
+            def cycles = this.zephyr.getProjectCycles(projectId, versionId)
+            testCycleId = cycles.find { it.value instanceof Map && it.value.name == cycleName }?.key
```
Comment (metmajer) [inline src/org/ods/usecase/JiraUseCaseZephyrSupport.groovy]: @hrcornejo I would assume that you would also need to check for the version, no?

**Comment by hrcornejo** [comment]
> 
> 
> @hrcornejo I have just tested this internally. Here's what I have found: the version shows UNDEFINED and the build number is not yet set. Let's walk through this together in the next days.

@metmajer It's strange, if "everything" fails, the version id must be "-1" --> UNSCHEDULED in Zephyr but no UNDEFINED

File: src/org/ods/usecase/JiraUseCaseZephyrSupport.groovy (hunk 1 above)
Comment (hrcornejo) [reply src/org/ods/usecase/JiraUseCaseZephyrSupport.groovy]: @metmajer the call `def versionId = this.getVersionId(projectId)` would must return always a valid id: 
- -1 (in default case --> UNSCHEDULED) 
- or a real id if it find it.

Perhaps is the util object not properly initialized in vars/phaseInit.groovy?
`new JiraUseCaseZephyrSupport(steps, jiraUseCase, registry.get(JiraZephyrService.class.name), registry.get(PipelineUtil.class.name))`

Comment (metmajer) [reply src/org/ods/usecase/JiraUseCaseZephyrSupport.groovy]: I'll have a check!

**Review by metmajer — COMMENTED** [review]
You are right @hrcornejo. I have mistyped this.

**Comment by metmajer** [comment]
Screenshot 2020-01-25 at 15 34 19

Screenshot 2020-01-25 at 15 35 46

**Pull request merged**